elector/elector.go

```diff
@@ -274,7 +281,16 @@ func (e *Elector) sendErr(err error) {
 
 // sendEvent sends the specified event on the events channel
 func (e *Elector) sendEvent(event Event) {
-	e.events <- event
+	select {
+	case e.events <- event:
+	case <-e.closing:
+		select {
```

Comment:
Why the inner select? If we've closed, there's no more delivery guarantee.

Reply:
you're right, there is no more delivery guarantee. this is my attempt at best-effort delivery. if you think this is too complex then we can probably get rid of it.